app/data/views.py

```diff
@@ -286,64 +305,73 @@ def crosstabs(self, request):
         try:
-            record_type = request.query_params['record_type']
+            record_type_id = request.query_params['record_type']
         except KeyError:
             raise ParseError(detail="The 'record_type' parameter is required")
         if param.endswith('period_type'):
-            return self._make_period_case(request.query_params[param], record_type)
+            return self._make_period_case(request.query_params[param], queryset)
         elif param.endswith('boundary_id'):
             return self._make_boundary_case(request.query_params[param])
         else:  # 'choices_path'; ensured by parent function
-            schema = RecordType.objects.get(pk=record_type).get_current_schema()
+            schema = RecordType.objects.get(pk=record_type_id).get_current_schema()
             return self._make_choices_case(schema, request.query_params[param].split(','))
 
-    def _make_period_case(self, period_type, record_type_id):
+    def _make_period_case(self, period_type, queryset):
         """Constructs a Django Case statement for a certain type of period.
 
         Args:
             period_type (string): One of 'hour', 'day', 'week_day', 'week', 'month', 'year'
-            record_type_id (string): Record type to use (uuid)
+            queryset (string): Filtered queryset to use for getting range for some period types
         Returns:
             (Case, labels), where Case is a Django Case object giving the period in which each
             record's occurred_from falls, and labels is a dict mapping Case values to period labels
 
         """
         easy_ranges = {  # Most date-related things are 1-indexed.
-            'month': xrange(1, 13),
-            'week': xrange(1, 54),  # Up to 53 weeks in a year
-            'week_day': xrange(1, 8),
-            'day': xrange(1, 32),
-            'hour': xrange(0, 24)
+            'month': {'keys': xrange(1, 13), 'labels': lambda x: calendar.month_name[x]},
+            # Up to 53 weeks in a year
+            'week': {'keys': xrange(1, 54), 'labels': lambda x: 'Week {}'.format(str(x))},
+            'week_day': {'keys': xrange(1, 8), 'labels': lambda x: calendar.day_name[x-1]},
+            'day': {'keys': xrange(1, 32), 'labels': lambda x: str(x)},
+            'hour': {'keys': xrange(0, 24), 'labels': lambda x: '{}:00'.format(x)},
         }
-        valid_periods = easy_ranges.keys() + ['year']
-        if period_type not in valid_periods:
-            raise ParseError(detail=('row_/col_period_type must be one of {}; received {}'
-                                     .format(valid_periods, period_type)))
 
-        # Set the range min / maxes based on the period type
-        period_range = None
+        # TODO: This sets year range based on the oldest and youngest records in the filtered
+        # queryset, but it (and possibly other date ranges) should be defined by the request's
+        # occurred_max and occurred_min parameters if available.
```

Comment:
Shouldn't the filtered queryset already be limited by the `occurred_max` and `occurred_min` query params by this point? I agree that determining the other ranges from the query set / query params is probably a good way to boost user friendliness. That way the number of days will be correct if the user does a report for a specific month, for example.